File: 01.Java-Basic/05.ConditionalStatementsAdvancedLab/src/SkiTrip.java
import java.util.Scanner;

public class SkiTrip {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int days = Integer.parseInt(scanner.nextLine());
        String room = scanner.nextLine();
        String rating = scanner.nextLine();

        int nights = days - 1;
        double nightPrice = 0;
        double discountPercentages = 0;
        switch (room) {
            case "room for one person":
                nightPrice = 18;
                break;
            case "apartment":
                nightPrice = 25;
                if (nights < 10) {
                    discountPercentages = 0.3;
                } else if (nights >= 10 && nights <= 15) {
                    discountPercentages = 0.35;
                } else {
                    discountPercentages = 0.5;
                }
                break;

            case "president apartment":
                nightPrice = 35;
                if (nights < 10) {
                    discountPercentages = 0.1;
                } else if (nights >= 10 && nights <= 15) {
                    discountPercentages = 0.15;
                } else {
                    discountPercentages = 0.2;
                }
                break;

            default:
                break;
        }

        double price = nights * nightPrice;
        if (discountPercentages > 0) {
            price = price - (price * discountPercentages);
        }

        if (rating.equals("positive")) {
            price = price + (price * 0.25);
        } else if (rating.equals("negative")) {
            price = price - (price * 0.1);
        }

        System.out.printf("%.2f", price);
    }
}

File: JavaScript/01.JavaScriptBasic/05Nested-Conditional-Statements-Lab/src/SkiTrip.js
function tripPrice(input) {
    let days = Number(input[0]);
    let room = input[1];
    let feedback = input[2];

    let nights = days - 1;
    let price = 0.0;
    let discount = 0.0;
    if (room === "room for one person") {
        price = 18.00;
    } else if (room === "apartment") {
        price = 25.00;
        if (days < 10) {
            discount = 0.30;
        } else if (days >= 10 && days <= 15) {
            discount = 0.35;
        } else {
            discount = 0.50;
        }
    } else {
        price = 35.00;
        if (days < 10) {
            discount = 0.10;
        } else if (days >= 10 && days <= 15) {
            discount = 0.15;
        } else {
            discount = 0.20;
        }
    }

    let totalPrice = nights * price;
    if (discount > 0) {
        totalPrice -= (totalPrice * discount);
    }

    if (feedback === "positive") {
        totalPrice += (totalPrice * 0.25);
    } else {
        totalPrice -= (totalPrice * 0.10);
    }

    console.log(`${totalPrice.toFixed(2)}`);
}

tripPrice(["14", "apartment", "positive"]);
tripPrice(["30", "president apartment", "negative"]);